types/chain_id.go

```diff
@@ -0,0 +1,24 @@
+package types
+
+import (
+	"io"
+
+	"github.com/bcicen/jstream"
```

Comment:
Could we add a todo here to use encoding/json/v2 when available?

Reply:
https://github.com/golang/go/discussions/63397
this one right?

Reply:
https://github.com/bcicen/jstream/pull/14
just patched the library to make api better, let's see if it's actively maintained or not.